src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -194,9 +195,16 @@ uct_ib_mlx5_iface_res_domain_init(uct_ib_mlx5_iface_res_domain_t *res_domain,
                       uct_ib_device_name(&md->dev));
             return UCS_ERR_IO_ERROR;
         }
+        status = uct_ib_mlx5_init_mmio_devx(md, &res_domain->uar,
+                                            UCT_IB_MLX5_MMIO_MODE_BF_POST);
+        if (status != UCS_OK) {
+            ibv_dealloc_td(res_domain->td);
```

Comment:
!! this is weird - why do we even need to use a thread/resource domain with devx? i would expect that we allocate UAR as a **replacement** for TD, not in addition to it

Reply:
Device supporting DEVX may create some objects via DEVX while some other via verbs. Hence it will need both DEVX UAR and TD.